pkg/bbgo/environment.go

```diff
@@ -139,8 +139,36 @@ func (environ *Environment) Connect(ctx context.Context) error {
 
 			marketDataStore := store.NewMarketDataStore(symbol)
 			marketDataStore.BindStream(session.Stream)
-
 			session.marketDataStores[symbol] = marketDataStore
+
+			standardIndicatorSet := NewStandardIndicatorSet(symbol)
+			standardIndicatorSet.BindMarketDataStore(marketDataStore)
+		}
+
+		// move market data store dispatch to here, use one callback to dispatch the market data
+		// session.Stream.OnKLineClosed(func(kline types.KLine) { })
+
+		now := time.Now()
+		for symbol := range loadedSymbols {
+			marketDataStore, ok := session.marketDataStores[symbol]
+			if !ok {
+				return errors.Errorf("symbol is not defined")
```

Comment:
It would be better to include `symbol` in the error string. That makes debugging much easier.

Reply:
make sense, thanks

Reply:
actually we add the store by the symbol, here we are iterating the loadedSymbols again, so it should be fine.